### PR TITLE
Ecm 89: Add optimistic locking for customer updates

### PR DESCRIPTION
**Pull request by ruben-burr**
As I am not familiar with most of the used technologies, someone should have a close look. In the meanwhile I will try to find time for implementing the missing tests.

### CODE REVIEW

**Comment by ruben-burr** [comment]
Tests for the optimistic lock failure in sequential transactions added. The optimistic lock failure in parallel transactions is not testable with acceptable effort in my opinion. Can be tested manually with a breakpoint in org.educama.customer.boundary.impl.CustomerBoundaryServiceImpl line 57.

**Comment by eberhardheber** [comment]
Hi Ruben,
can you rebase this PR. Because it has conflicts.
Greetings
Eberhard

File: backend/pom.xml [hunk 1]
```diff
@@ -95,6 +101,9 @@
                         to produce a separate artifact that is suitable for use as a dependency.
                         See http://docs.spring.io/spring-boot/docs/1.4.x/reference/htmlsingle/#howto-create-an-additional-executable-jar -->
                     <classifier>exec</classifier>
+                    <jvmArguments>
+                        -Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005
```
Comment (eberhardheber) [inline backend/pom.xml]: Why do we need these arguments?

File: backend/src/main/java/org/educama/customer/api/CustomerController.java [hunk 2]
```diff
@@ -47,7 +50,7 @@
     @RequestMapping(method = RequestMethod.POST)
     public ResponseEntity<CustomerResource> createCustomer(
             @Valid @RequestBody SaveCustomerResource saveCustomerResource) {
-
+    	logger.info(()-> "createCustomer called with " + saveCustomerResource);
```
Comment (eberhardheber) [inline backend/src/main/java/org/educama/customer/api/CustomerController.java]: It is possible to check the log level? If we set the level in the future to warning or other higher level the string concatenation will be excuted.

File: backend/src/main/java/org/educama/customer/api/resource/SaveCustomerResource.java [hunk 3]
```diff
@@ -10,8 +10,15 @@
 
     @NotEmpty
     public String name;
+    
+    public long version;
```
Comment (eberhardheber) [inline backend/src/main/java/org/educama/customer/api/resource/SaveCustomerResource.java]: Does this field need a @NotNull annotation?

**Comment by larsduelfer** [comment]
Will not be integrated at the moment.

**Pull request closed without merging**